DXMainClient/DXGUI/Multiplayer/MapTextureCacheManager.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+using DTAClient.Domain.Multiplayer;
+
+using SixLabors.ImageSharp;
+
+namespace DTAClient.DXGUI.Multiplayer
+{
+    public class MapTextureCacheManager : IDisposable
+    {
+        public const int MaxCacheSize = 100;
+        public const int SleepIntervalMS = 100;
+
+        private readonly ConcurrentDictionary<Map, Image> mapTextures = [];
+
+        private readonly ConcurrentDictionary<Map, byte> missedMaps = [];
+
+        private readonly CancellationTokenSource cancellationTokenSource = new();
+
+        public MapTextureCacheManager() =>
+            Task.Run(() => MapTextureLoadingService(cancellationTokenSource.Token));
+
+        public void Dispose() =>
+            cancellationTokenSource?.Cancel();
+
+        public Image GetMapTextureIfAvailable(Map map)
+        {
+            if (mapTextures.TryGetValue(map, out Image image))
+                return image;
+
+            if (missedMaps.Count < MaxCacheSize)
+                missedMaps.TryAdd(map, 0);
+
+            return null;
+        }
+
+        private async Task MapTextureLoadingService(CancellationToken cancellationToken)
```

Comment:
don't think the functions should be called like so, the functions is "do something", not "a thing"

Reply:
will fix it